### PR TITLE
Transformations: Adds labels as columns transformer

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
I think this covers some cases, but I might have missed something. Should this work for 2 data frames with the same time like the example below?

```ts
    const seriesA = toDataFrame({
      name: 'A',
      fields: [
        { name: 'time', type: FieldType.time, values: [1000] },
        { name: 'temp', type: FieldType.number, values: [1], labels: { location: 'inside', feelsLike: 'ok' } },
      ],
    });

    const seriesB = toDataFrame({
      name: 'B',
      fields: [
        { name: 'time', type: FieldType.time, values: [1000] },
        { name: 'temp', type: FieldType.number, values: [-1], labels: { location: 'outside', sky: 'cloudy' } },
      ],
    });
```
and should the expected fields be:

```ts
    const expected: Field[] = [
      { name: 'time', type: FieldType.time, values: new ArrayVector([1000]), config: {} },
      { name: 'location {A}', type: FieldType.string, values: new ArrayVector(['inside']), config: {} },
      { name: 'location {B}', type: FieldType.string, values: new ArrayVector(['outside']), config: {} },
      { name: 'feelsLike', type: FieldType.string, values: new ArrayVector(['ok']), config: {} },
      { name: 'sky', type: FieldType.string, values: new ArrayVector(['cloudy']), config: {} },
      { name: 'temp {A}', type: FieldType.number, values: new ArrayVector([1]), config: {} },
      { name: 'temp {B}', type: FieldType.number, values: new ArrayVector([-1]), config: {} },
    ];
```

**Which issue(s) this PR fixes**:
Relates #23424

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ryantxu** [comment]
@kylebrandt -- thoughts?

**Comment by torkelo** [comment]
My thoughts would be that is would be two rows as the labels are not the same , so the field name should not be location (A) etc but just location

**Comment by hugohaggmark** [comment]
I've added support for frames with same timestamp but not sure the new code covers any the special cases. Any feedback would be highly appreciated 🙏

**Comment by torkelo** [comment]
Can you add a UI registry item for this as well? example:
https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/TransformersUI/ReduceTransformerEditor.tsx

But with an empty UI editor (no options). That way we can test this with Prometheus queries & graphite queries

**Comment by hugohaggmark** [comment]
> Can you add a UI registry item for this as well? example:
> https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/TransformersUI/ReduceTransformerEditor.tsx
> 
> But with an empty UI editor (no options). That way we can test this with Prometheus queries & graphite queries

Done @torkelo

**Comment by hugohaggmark** [comment]
I've noticed that the resulting frame doesn't sort time values in order. I'm looking into this.

**Comment by hugohaggmark** [comment]
So I've sorted out sorting :) but and I could probably make this more efficient but I'll wait for more feedback before I do anything more with this.

**Comment by torkelo** [comment]
It is looking promising, but we have a fundamental problem that is not in this transform but how we transform time series to DataFrame. 

Graphite & Prometheus already generates series name based on metric name and series name. this is then transformed using toDataFrame to a DataFrame. In this transform we are using the series name as the field name for the value. So this is a big problem for this transformer as we end up with this: 

Screenshot from 2020-04-22 09-04-45

Exact same problem:

Screenshot from 2020-04-22 09-17-48

@ryantxu, as you mentioned, think we need to rethink a bit how we use labels, specifically  in terms of data frame and field name

**Comment by hugohaggmark** [comment]
> It is looking promising, but we have a fundamental problem that is not in this transform but how we transform time series to DataFrame.
> 
> Graphite & Prometheus already generates series name based on metric name and series name. this is then transformed using toDataFrame to a DataFrame. In this transform we are using the series name as the field name for the value. So this is a big problem for this transformer as we end up with this:
> 
> Screenshot from 2020-04-22 09-04-45
> 
> Exact same problem:
> 
> Screenshot from 2020-04-22 09-17-48
> 
> @ryantxu, as you mentioned, think we need to rethink a bit how we use labels, specifically in terms of data frame and field name

Yes, I've noticed this too. Is there something I should do in this transformer?

File: packages/grafana-data/src/transformations/transformers.ts
Comment (torkelo) [inline packages/grafana-data/src/transformations/transformers.ts:27]: can we rename it to labelsToFieldsTransformer?

**Review by torkelo — APPROVED** [review]
It looks good. I opened a follow-up issue for the field name problem: https://github.com/grafana/grafana/issues/23769  

But please rename the transformer so it matches the UI transformer name, and do any optimizations you  where thinking about.

File: packages/grafana-data/src/transformations/transformers.ts
Comment (hugohaggmark) [reply packages/grafana-data/src/transformations/transformers.ts]: Rename done!

**Comment by hugohaggmark** [comment]
I will merge this as is, if there's time before beta I'll do another optimization pass.

**Pull request merged**
